invenio/utils/text.py

```diff
@@ -737,7 +787,8 @@ def strip_accents(x):
 
 
 def slugify(text, delim=u'-'):
-    """Generates an ASCII-only slug."""
+    """Generate an ASCII-only slug."""
+    text = unicode(text)
```

Comment:
@hachreak it would be better to encode the input in upgrade recipe where you are sure that you have UTF8 string.